odoo/tools/misc.py

```diff
@@ -1335,20 +1335,36 @@ def format_time(env, value, tz=False, time_format='medium', lang_code=False):
 
         :param value: the time to format
         :type value: `datetime.time` instance. Could be timezoned to display tzinfo according to format (e.i.: 'full' format)
-        :param format: one of “full”, “long”, “medium”, or “short”, or a custom date/time pattern
+        :param tz: name of the timezone  in which the given datetime should be localized
+        :param time_format: one of “full”, “long”, “medium”, or “short”, or a custom time pattern
         :param lang_code: ISO
 
         :rtype str
+
     """
+
     if not value:
         return ''
 
+    if isinstance(value, datetime.time):
+        localized_datetime = value
+    else:
+        if isinstance(value, str):
+            value = odoo.fields.Datetime.from_string(value)
+        tz_name = tz or env.user.tz or 'UTC'
+        utc_datetime = pytz.utc.localize(value, is_dst=False)
+        try:
+            context_tz = pytz.timezone(tz_name)
+            localized_datetime = utc_datetime.astimezone(context_tz)
+        except Exception:
+            localized_datetime = utc_datetime
+
     lang = get_lang(env, lang_code)
-    locale = babel_locale_parse(lang.code)
+    locale = babel_locale_parse(lang.code or lang_code)  # lang can be inactive, so `lang`is empty
```

Comment:
`get_lang` will always return a value that is in `get_installed` so you should always get a value, no?

Reply:
That piece of code was inspired from the code in the format_datetime method, after taking a deeper look at this, it appears that `get_lang` always returns a value and that this verification is not necessary, shouldn't we change this too for format_datetime?

Reply:
> shouldn't we change this too for format_datetime?

Not necessarily, if you are not modifying it. This is not a bug, just a bit useless code. Can be cleaned but not a big deal to leave it. Was added at 1b24f60a5ce26cebe9ae7306a01c02ac4c0f420f, before the `get_lang`

Reply:
Okey then, no need to modify it.